File: pixels-core/src/main/java/cn/edu/ruc/iir/pixels/core/exception/PixelsFileVersionInvalidException.java
package cn.edu.ruc.iir.pixels.core.exception;

import cn.edu.ruc.iir.pixels.core.PixelsVersion;

/**
 * pixels
 *
 * @author guodong
 */
public class PixelsFileVersionInvalidException
        extends PixelsRuntimeException
{
    private static final long serialVersionUID = -6176420971384589629L;

    public PixelsFileVersionInvalidException(int version)
    {
        super(String.format("This is not a valid file version %d for current reader %d",
                        version, PixelsVersion.currentVersion().getVersion()));
    }
}

File: writer_cpp_test/pixels-core/lib/exception/PixelsFileVersionInvalidException.cpp
//
// Created by liyu on 3/13/23.
//

#include "exception/PixelsFileVersionInvalidException.h"

PixelsFileVersionInvalidException::PixelsFileVersionInvalidException(uint32_t version) {
    std::cout<<"This is not a valid file version "
        << version
        <<" for current reader "
        << PixelsVersion::currentVersion()
        <<std::endl;
}
